src/crawler/crawlers/arso_daily_weather/crawler.js

```diff
@@ -0,0 +1,247 @@
+                let i = 0;
+                for (const stationSensors of Object.entries(station)) {
+                    const sen = sensors.filter((x) => x.uuid === stationSensors[0]);
+
+                    if (sen.length === 0) {
+                        if (!stationSensors[0].includes('Datum') && !stationSensors[0].includes('Postaja')) {
+                            await this.SQLUtils.insertSensor(`${station.Postaja} - ${i}`, station.Postaja, stationSensors[0], `${station.Postaja} - ${stationSensors[0]}`);
+                            i++;
+                        }
+                    }
+                }
+            }
+
+            let val;
+            let date;
+            let sensorUUID;
+
+            // updating sensors array with new inserts
+            sensors = await this.SQLUtils.getSensors();
+
+            // inserting measurements
+            for (const stationSensors of Object.entries(station)) {
+                if (stationSensors[0].includes('Datum')) {
+                    date = stationSensors[1];
+                } else if (!stationSensors[0].includes('Datum') && !stationSensors[0].includes('Postaja')) {
+                    sensorUUID = `${station.Postaja} - ${el}`;
+                    val = stationSensors[1];
+                    el++;
+
+                    for (const sensorMeta of sensors) {
+                        if ((sensorUUID === sensorMeta.uuid) && (val !== String.fromCharCode(160))) {
+                            if (isNaN(val)) val = this.stringDataConversion(val);
+
+                            try {
+                                const SQL = this.SQLUtils.insertMeasurementSQL(sensorMeta.id, moment(date).unix(), val);
+                                // console.log(SQL)
+
+                                if (moment(date).unix() !== this.lastUnix) {
+                                    this.SQLUtils.processSQL(SQL);
+                                }
+                            } catch (e) {console.log(SQL)}
+                        }
+                    }
+                }
+            };
+        }
+
+
+        // write final state
+        CrawlerUtils.saveState(__dirname, this.state);
+
+        console.log('Finishing crawl: ' + this.config.id);
+    };
+
+    async getData(url = this.config.start) {
+        const weatherData = [];
+
+        const body = await CrawlerUtils.getURL(url);
+        const rawMeasurements = [];
+        const rawHeads = [];
+        const rawStyleHtml = [];
+        const $ = cheerio.load(body);
+
+        $('.meteoSI-table tbody tr td').each((i, el) => {
+            const text = $(el).text();
+            rawMeasurements.push(text);
+            rawStyleHtml.push($(el).html());
+        });
+        $('.meteoSI-table thead tr th').each((i, el) => {
+            const text = $(el).text();
+            rawHeads.push(text);
+        });
+
+        const measurements = [];
+        while (rawMeasurements.length) measurements.push(rawMeasurements.splice(0, 11));
```

Comment:
Write in a longer form - it is easier to read. Also comment, what splice does. It is not intuitive.

while (..) {
  ...
}